iac-extensions/arm/src/main/java/org/sonar/iac/arm/parser/bicep/TreeFactory.java

```diff
@@ -476,6 +479,18 @@ public AmbientTypeReference ambientTypeReference(SyntaxToken token) {
     return new AmbientTypeReferenceImpl(token);
   }
 
+  public TypeExpressionAble typeReference(TypeExpressionAble type, Optional<List<TypeReferenceSuffix>> optionalSuffixes) {
+    List<TypeReferenceSuffix> suffixes = optionalSuffixes.or(List.of());
+    for (TypeReferenceSuffix suffix : suffixes) {
+      type = suffix.setType(type);
```

Comment:
I would change the method name to not look like a setter.
`setType` name made me initially think it was a simple setter, I didn't expect it to instantiate an object nor to return something.

Reply:
Totally agree, I'm trying to come up with a better name but didn't settle on one I like 😆